File: src/TradingSystem.Web/Services/NewsApiService.cs
using System.Net.Http.Json;
using TradingSystem.Web.Models;

namespace TradingSystem.Web.Services;

public interface INewsApiService
{
    Task<NewsAnalysisResponse?> AnalyzeAsync(NewsAnalysisRequest request);
    Task<List<NewsItem>?> GetNewsAsync(string symbol);
}

public class NewsApiService : INewsApiService
{
    private readonly HttpClient _http;

    public NewsApiService(HttpClient http)
    {
        _http = http;
    }

    public async Task<NewsAnalysisResponse?> AnalyzeAsync(NewsAnalysisRequest request)
    {
        var response = await _http.PostAsJsonAsync("news/analyze", request);
        var result = await response.Content.ReadFromJsonAsync<ApiResponse<NewsAnalysisResponse>>();
        return result?.Data;
    }

    public async Task<List<NewsItem>?> GetNewsAsync(string symbol)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<List<NewsItem>>>($"news/{symbol}");
        return response?.Data;
    }
}

File: src/TradingSystem.Angular/src/app/services/news-api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import { ApiResponse, NewsItem, NewsAnalysisRequest, NewsAnalysisResponse } from '../models/models';

@Injectable({ providedIn: 'root' })
export class NewsApiService {
  constructor(private http: HttpClient) {}

  getNews(symbol: string): Observable<NewsItem[]> {
    return this.http.get<ApiResponse<NewsItem[]>>(`/news/${symbol}`).pipe(
      map(r => r.data || []),
      catchError(() => of([]))
    );
  }

  analyze(request: NewsAnalysisRequest): Observable<NewsAnalysisResponse | null> {
    return this.http.post<ApiResponse<NewsAnalysisResponse>>('/news/analyze', request).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }
}
